### PR TITLE
Remove first_router and last_routers consolidation strategies

### PR DESCRIPTION
**Pull request by Mallets**
Removes unimplemented consolidation strategies.

### CODE REVIEW

File: include/zenoh-pico/api/constants.h [hunk 1]
```diff
@@ -133,24 +135,24 @@ typedef enum
...
 } z_queryable_kind_t;
 
 /**
  * Consolidation mode values.
  *
  * Enumerators:
- *     Z_CONSOLIDATION_MODE_FULL: Guaranties unicity of replies. Optimizes bandwidth.
- *     Z_CONSOLIDATION_MODE_LAZY: Does not garanty unicity. Optimizes latency.
+ *     Z_CONSOLIDATION_MODE_LAST_VALUE: Guaranties unicity of replies. Optimizes bandwidth.
+ *     Z_CONSOLIDATION_MODE_MONOTONIC: Does not garanty unicity. Optimizes latency.
```
Comment (cguimaraes) [inline include/zenoh-pico/api/constants.h:148]: I suggest to have the order in the comments to be the same as in the enum definition.

Comment (Mallets) [reply include/zenoh-pico/api/constants.h]: Fixed and improved docs.

File: include/zenoh-pico/api/constants.h [hunk 2]
```diff
@@ -133,24 +135,25 @@ typedef enum
...
 /**
  * Consolidation mode values.
  *
  * Enumerators:
- *     Z_CONSOLIDATION_MODE_FULL: Guaranties unicity of replies. Optimizes bandwidth.
- *     Z_CONSOLIDATION_MODE_LAZY: Does not garanty unicity. Optimizes latency.
- *     Z_CONSOLIDATION_MODE_NONE: No consolidation.
+ *     Z_CONSOLIDATION_MODE_NONE: No consolidation is applied. Replies may come in any order and any number.
+ *     Z_CONSOLIDATION_MODE_MONOTONIC: It guarantees that any reply will be monotonic in time w.r.t.
+ *          the previous received replies. It does not garantuee unicity of replies. Optimizes latency.
```
Comment (p-avital) [inline include/zenoh-pico/api/constants.h:149]: Does it not guarantee unicity of replies with the same key-timestamp tuple? I think zenoh-rs does, and if not, it's an easy fix that shouldn't change the performances :)

File: include/zenoh-pico/api/constants.h [hunk 3]
```diff
@@ -133,24 +135,25 @@ typedef enum
...
+ *     Z_CONSOLIDATION_MODE_LAST_VALUE: It guarantees unicity of replies. Optimizes bandwidth.
  */
 typedef enum
 {
     Z_CONSOLIDATION_MODE_NONE = 0,
-    Z_CONSOLIDATION_MODE_LAZY = 1,
-    Z_CONSOLIDATION_MODE_FULL = 2
+    Z_CONSOLIDATION_MODE_MONOTONIC = 1,
+    Z_CONSOLIDATION_MODE_LAST_VALUE = 2,
```
Comment (p-avital) [inline include/zenoh-pico/api/constants.h:156]: As you said in DM, I agree that `LATEST_VALUE` is better.

File: include/zenoh-pico/api/types.h [hunk 4]
```diff
@@ -338,20 +336,20 @@ typedef _z_reply_data_t z_reply_data_t;
  */
 typedef _z_reply_t z_reply_t;
 
-#define _TYPEDEF_ARRAY(type, alias, elem, name)                               \
-    typedef type alias;                                                       \
-    static inline elem *z_##name##_array_get(const alias *a, size_t k)        \
```
Comment (p-avital) [inline include/zenoh-pico/api/types.h:343]: I just noticed, could you take Carlos's formatter settings? They should be `"C_Cpp.clang_format_fallbackStyle": "{ BasedOnStyle: Google, IndentWidth: 4, ColumnLimit: 120}"` in VSCode's configuration :)

File: include/zenoh-pico/api/primitives.h [hunk 5]
```diff
@@ -349,17 +349,7 @@ z_query_consolidation_t z_query_consolidation_default(void);
...
-/**
- * Full consolidation performed on last router and at reception.
- *
- * This strategy offers a good latency while optimizing bandwidth on the last transport link between the router and the application.
- *
- * Returns:
- *   Returns the constructed :c:type:`z_query_consolidation_t`.
- */
-z_query_consolidation_t z_query_consolidation_last_router(void);
+z_query_consolidation_t z_query_consolidation_last_value(void);
```
Comment (p-avital) [inline include/zenoh-pico/api/primitives.h:352]: Same as the enum value, let's go for "latest" :)

**Review by p-avital — CHANGES_REQUESTED** [review]

File: include/zenoh-pico/api/constants.h (hunk 2 above)
Comment (Mallets) [reply include/zenoh-pico/api/constants.h:149]: It works exactly as in `zenoh-rs`. It seems it is simply poorly worded. What about: 
> Z_CONSOLIDATION_MODE_MONOTONIC: It guarantees that any reply for a given key expression will be monotonic in time w.r.t. the previous received replies for the same key expression. I.e., for a key expression A it is guaranteed that two replies received at t1 and t2 will have timestamp ts2 > ts1.  It optimizes latency.

File: include/zenoh-pico/api/constants.h [hunk 6]
```diff
@@ -40,13 +41,12 @@ typedef enum
  *     Z_KEYEXPR_CANON_SINGLE_STAR_AFTER_DOUBLE_STAR: The key contains ``** / *``, which must be replaced by ``* / **``.
  *     Z_KEYEXPR_CANON_DOUBLE_STAR_AFTER_DOUBLE_STAR: The key contains ``** / **``, which must be replaced by ``**``.
  *     Z_KEYEXPR_CANON_EMPTY_CHUNK: The key contains empty chunks.
- *     Z_KEYEXPR_CANON_STARS_IN_CHUNK: The key contains a ``*`` in a chunk without being escaped by a DSL, which is forbidden.
- *     Z_KEYEXPR_CANON_DOLLAR_AFTER_DOLLAR_OR_STAR: The key contains ``$*$`` or ``$$``, which is forbidden.
+ *     Z_KEYEXPR_CANON_STARS_IN_CHUNK: The key contains a ``*`` in a chunk without being escaped by a DSL, which is
+ * forbidden. Z_KEYEXPR_CANON_DOLLAR_AFTER_DOLLAR_OR_STAR: The key contains ``$*$`` or ``$$``, which is forbidden.
```
Comment (p-avital) [inline include/zenoh-pico/api/constants.h:45]: Missing a newline here :)

File: include/zenoh-pico/api/primitives.h (hunk 5 above)
Comment (p-avital) [reply include/zenoh-pico/api/primitives.h:364]: Still not fully renamed

**Review by p-avital — APPROVED** [review]

**Pull request merged**